clang/lib/Headers/avx10_2satcvtdsintrin.h

```diff
@@ -0,0 +1,453 @@
+  ((unsigned long long)__builtin_ia32_vcvttssd2usi64((__v2df)(__m128d)(A),     \
+                                                     (const int)(R)))
+#endif
+
+#define _mm_cvtt_roundsss_i32(A, R)                                            \
+  ((int)__builtin_ia32_vcvttsss2si32((__v4sf)(__m128)(A), (const int)(R)))
+
+#define _mm_cvtt_roundsss_si32(A, R)                                           \
+  ((int)__builtin_ia32_vcvttsss2si32((__v4sf)(__m128)(A), (const int)(R)))
+
+#ifdef __x86_64__
+#define _mm_cvtt_roundsss_i64(A, R)                                            \
+  ((long long)__builtin_ia32_vcvttsss2si64((__v4sf)(__m128)(A), (const int)(R)))
+
+#define _mm_cvtt_roundsss_si64(A, R)                                           \
+  ((long long)__builtin_ia32_vcvttsss2si64((__v4sf)(__m128)(A), (const int)(R)))
+#endif
+
+#define _mm_cvtt_roundsss_u32(A, R)                                            \
+  ((unsigned int)__builtin_ia32_vcvttsss2usi32((__v4sf)(__m128)(A),            \
+                                               (const int)(R)))
+
+#ifdef __x86_64__
+#define _mm_cvtt_roundsss_u64(A, R)                                            \
+  ((unsigned long long)__builtin_ia32_vcvttsss2usi64((__v4sf)(__m128)(A),      \
+                                                     (const int)(R)))
+#endif
+
+//  128 Bit : Double -> int
+#define _mm_cvttspd_epi32(A)                                                   \
+  ((__m128i)__builtin_ia32_vcvttpd2dqs128_mask(                                \
+      (__v2df)(__m128d)A, (__v4si)(__m128i)_mm_undefined_si128(),              \
+      (__mmask8)(-1)))
+
+#define _mm_mask_cvttspd_epi32(W, U, A)                                        \
+  ((__m128i)__builtin_ia32_vcvttpd2dqs128_mask(                                \
+      (__v2df)(__m128d)A, (__v4si)(__m128i)W, (__mmask8)U))
+
+#define _mm_maskz_cvttspd_epi32(U, A)                                          \
+  ((__m128i)__builtin_ia32_vcvttpd2dqs128_mask(                                \
+      (__v2df)(__m128d)A, (__v4si)(__m128i)_mm_setzero_si128(), (__mmask8)U))
+
+//  256 Bit : Double -> int
+static __inline__ __m128i __DEFAULT_FN_ATTRS _mm256_cvttspd_epi32(__m256d A) {
+  return ((__m128i)__builtin_ia32_vcvttpd2dqs256_round_mask(
+      (__v4df)(__m256d)A, (__v4si)_mm_undefined_si128(), (__mmask8)-1,
+      _MM_FROUND_CUR_DIRECTION));
+}
+
+static __inline__ __m128i __DEFAULT_FN_ATTRS
+_mm256_mask_cvttspd_epi32(__m128i W, __mmask8 U, __m256d A) {
+  return ((__m128i)__builtin_ia32_vcvttpd2dqs256_round_mask(
+      (__v4df)A, (__v4si)W, U, _MM_FROUND_CUR_DIRECTION));
+}
+
+static __inline__ __m128i __DEFAULT_FN_ATTRS
+_mm256_maskz_cvttspd_epi32(__mmask8 U, __m256d A) {
+  return ((__m128i)__builtin_ia32_vcvttpd2dqs256_round_mask(
+      (__v4df)A, (__v4si)_mm_setzero_si128(), U, _MM_FROUND_CUR_DIRECTION));
+}
+
+#define _mm256_cvtts_roundpd_epi32(A, R)                                       \
```

Comment:
Naming convention needs to be coherence. Choose either put `s` behind `cvtt` or `cvtt_round`.